src/test/java/tests/SortQualifierTests.java

```diff
@@ -1,27 +1,74 @@
 package tests;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-
-import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.stream.Collectors;
-
-import org.junit.Test;
-
 import backend.interfaces.IModel;
-import backend.resource.Model;
-import backend.resource.TurboIssue;
-import backend.resource.TurboLabel;
+import backend.resource.*;
 import filter.Parser;
 import filter.expression.FilterExpression;
 import filter.expression.Qualifier;
+import org.junit.Test;
+import prefs.Preferences;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class SortQualifierTests {
+    private class ModelStub extends MultiModel {
+        List<TurboIssue> issues;
+        List<TurboMilestone> milestones = createSampleMilestone();
+
+        public ModelStub(Preferences prefs, List<TurboIssue> issues) {
+            super(prefs);
+            this.issues = issues;
+
+            issues.get(0).setMilestone(milestones.get(0));
+            issues.get(1).setMilestone(milestones.get(1));
+            issues.get(2).setMilestone(milestones.get(2));
+            issues.get(3).setMilestone(milestones.get(3));
+        }
+
+        @Override
+        public synchronized List<TurboIssue> getIssues() {
+            return issues;
+        }
+
+        @Override
+        public synchronized List<TurboMilestone> getMilestones() {
+            return milestones;
+        }
+
+        @Override
+        public Optional<TurboMilestone> getMilestoneOfIssue(TurboIssue issue) {
+            if (!issue.getMilestone().isPresent()) {
+                return Optional.empty();
+            }
+
+            Integer id = issue.getMilestone().get();
+
+            for (TurboMilestone milestone : getMilestones()) {
+                if (milestone.getId() == id) {
+                    return Optional.of(milestone);
+                }
+            }
+            return Optional.empty();
```

Comment:
I just take a look at Mockito, looks pretty neat and could be exactly what we need.